main/main.go

```diff
@@ -169,13 +169,25 @@ func main() {
 	}
 
 	// Initialize the database
-	db, err := database.NewJSONFileStorage(logger, cfg.StorageLocation, database.GetConfigRelayerIDs(&cfg))
-	if err != nil {
-		logger.Error(
-			"Failed to create database",
-			zap.Error(err),
-		)
-		panic(err)
+	var db database.RelayerDatabase
+	if cfg.RedisURL != "" {
```

Comment:
can we create a new method `NewDatabase` or `NewRelayerDatabase`, pass in the config there, and perform this logic there? The function can return the interface `RelayerDatabase` type

Reply:
Good call. Added.